torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -16,6 +18,28 @@
 aten = torch.ops.aten
 
 
+# Some fusions would only work on Xeon SP processors gen 2 & above
+def is_avx512_vnni_supported():
+    if sys.platform != "linux":
+        return False
+    with open("/proc/cpuinfo", encoding="ascii") as f:
+        lines = f.read()
+    return "vnni" in lines
+
+
+# Some BF16 fusions would only work on Xeon SP processors gen 3 (Cooper Lake) & above
+def is_avx512_bf16_supported():
+    if sys.platform != "linux":
+        return False
+    with open("/proc/cpuinfo", encoding="ascii") as f:
+        lines = f.read()
+    return "avx512_bf16" in lines
+
+
+IS_AVX512_VNNI_SUPPORTED = is_avx512_vnni_supported()
+IS_AVX512_BF16_SUPPORTED = is_avx512_bf16_supported()
```

Comment:
importing should not have the side effect of reading files.  Move these to utils and put `lru_cache` on them to compute lazily.

Reply:
Will do. Thanks!